src/rules.ts

```diff
@@ -170,7 +170,8 @@ const br = /^( {2,}|\\)\n(?!\s*$)/;
 const inlineText = /^(`+|[^`])(?:(?= {2,}\n)|[\s\S]*?(?:(?=[\\<!\[`*_]|\b_|$)|[^ ](?= {2,}\n)))/;
 
 // list of unicode punctuation marks, plus any missing characters from CommonMark spec
-const _punctuation = '\\p{P}\\p{S}';
+// fallback to explicit punctuation list if runtime environment doesn't support punctuation regex syntax
+const _punctuation = isSupportPunctuationRegex ? '\\p{P}\\p{S}' : '!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~';
```

Comment:
These two regular expressions are very different which means markdown will render differently in different browsers.

You should be able to just change the string to a regular expression. Then babel can transpile it correctly.

```diff
- const _punctuation = '\\p{P}\\p{S}';
+ const _punctuation = /\p{P}\p{S}/u;
```